Update the setting defaults in sql plugin (#11598)
opensearch-project/documentation-website: _search-plugins/sql/settings.md

Setting | Default | Description
:--- | :--- | :---
`plugins.sql.enabled` | `true` | Change to `false` to disable the `SQL` support in the plugin.
`plugins.ppl.enabled` | `true` | Change to `false` to disable the `PPL` support in the plugin.
`plugins.sql.slowlog` | `2` | Configures the time limit (in seconds) for slow queries. The plugin logs slow queries as `Slow query: elapsed=xxx (ms)` in `opensearch.log`.

> `plugins.sql.cursor.keep_alive` | `1m` | Configures the amount of time that the cursor context remains open. Because cursor contexts are resource-intensive, we recommend a low value.

Reviewer: ```suggestion
`plugins.sql.cursor.keep_alive` | `1m` | Configures the amount of time that the cursor context remains open. Because cursor contexts are resource intensive, we recommend a low value.
```

`plugins.calcite.pushdown.rowcount.estimation.factor` | `0.9` | A factor used to multiply the row count of a table scan to estimate the resulting row count. We recommend using the default value.
`plugins.calcite.all_join_types.allowed` | `false` | Enables performance-sensitive join types, such as `RIGHT`, `FULL`, and `CROSS` joins. Change to `true` to allow these join operations.
`plugins.ppl.syntax.legacy.preferred` | `true` | Controls PPL syntax behaviors, including default argument values. When `false`, uses newer syntax standards. For more information, see the [legacy syntax documentation](https://github.com/opensearch-project/sql/blob/main/docs/user/ppl/admin/settings.rst#plugins-ppl-syntax-legacy-preferred).
`plugins.ppl.values.max.limit` | `0` | Sets the maximum number of unique values that the `VALUES` aggregation function can return. A value of `0` indicates no limit.
`plugins.ppl.rex.max_match.limit` | `10` | Sets the maximum number of matches extracted by the `rex` command.
`plugins.ppl.subsearch.maxout` | `10000` | Sets the maximum number of rows to return from a subsearch.

> `plugins.ppl.join.subsearch_maxout` | `50000` | Sets the maximum number of rows from to return from a subsearch used in a join operation.

Reviewer: ```suggestion
`plugins.ppl.join.subsearch_maxout` | `50000` | Sets the maximum number of rows to return from a subsearch used in a join operation.
```

`plugins.ppl.pattern.method` | `simple_pattern` | Sets the method for the `patterns` command. Valid values are `simple_pattern` and `brain`. For more information, see [`patterns` syntax](https://github.com/opensearch-project/sql/blob/main/docs/user/ppl/cmd/patterns.rst#syntax).
`plugins.ppl.pattern.mode` | `label` | Sets the mode for the `patterns` command. Valid values are `label` and `aggregation`. For more information, see [`patterns` syntax](https://github.com/opensearch-project/sql/blob/main/docs/user/ppl/cmd/patterns.rst#syntax).
`plugins.ppl.pattern.max.sample.count` | `10` | Sets the maximum number of sample logs returned per pattern in aggregation mode.
`plugins.ppl.pattern.buffer.limit` | `100000` | Sets the size of the internal temporary buffer used by the `brain` algorithm.
`plugins.ppl.pattern.show.numbered.token` | `false` | Change to `true` to enable the numbered token output format.

> `plugins.ppl.query.timeout` | `5m` | Configures the maximum amount of time a PPL query can run. If a query exceeds this limit, execution stops and returns a timeout error.

Reviewer: ```suggestion
`plugins.ppl.query.timeout` | `5m` | Configures the maximum amount of time that a PPL query can run. If a query exceeds this limit, execution stops and returns a timeout error.
```

Setting | Default | Description
:--- | :--- | :---
`plugins.sql.enabled` | `true` | Change to `false` to disable the `SQL` support in the plugin.
`plugins.ppl.enabled` | `true` | Change to `false` to disable the `PPL` support in the plugin.
`plugins.sql.slowlog` | `2` | Configures the time limit (in seconds) for slow queries. The plugin logs slow queries as `Slow query: elapsed=xxx (ms)` in `opensearch.log`.

> `plugins.sql.cursor.keep_alive` | `1m` | Configures how long the cursor context is kept open. Cursor contexts are resource-intensive, so we recommend a low value.

Reviewer 2: ```suggestion
`plugins.sql.cursor.keep_alive` | `1m` | Configures the amount of time that the cursor context remains open. Because cursor contexts are resource-intensive, we recommend a low value.
```

`plugins.query.memory_limit` | `85%` | Configures the heap memory usage limit for the circuit breaker of the query engine.

> `plugins.query.size_limit` | `10000` | Sets the maximum amount of rows returned from a query execution results.

Reviewer 2: ```suggestion
`plugins.query.size_limit` | `10000` | Sets the maximum number of rows returned from a query execution.
```

`plugins.calcite.enabled` | `true` | Enables the Apache Calcite query engine, including advanced SQL and PPL capabilities such as subsearch, join, and lookup operations.
`plugins.calcite.pushdown.enabled` | `true` | Change to `false` to disable the operator push-down optimization. We recommend using the default value.
`plugins.calcite.fallback.allowed` | `false` | Change to `true` to allow fallback to v2 engine.
`plugins.calcite.pushdown.rowcount.estimation.factor` | `0.9` | The value is a factor to multiply the row count of the table scan to get the estimated row count. We recommend using the default value.
`plugins.calcite.all_join_types.allowed` | `false` | Enables performance-sensitive join types, like `RIGHT`, `FULL`, and `CROSS` joins. Change to `true` to allow these join operations.
`plugins.ppl.syntax.legacy.preferred` | `true` | Controls certain PPL syntax behaviors, including default argument values. When `false`, uses newer syntax standards.
`plugins.ppl.values.max.limit` | `0` | Sets the maximum number of unique values that the `VALUES` aggregation function can return. A value of `0` indicates no limit.

> `plugins.ppl.rex.max_match.limit` | `10` | Maximum number of matches to extract in `rex` command.

Reviewer 2: ```suggestion
`plugins.ppl.rex.max_match.limit` | `10` | Sets the maximum number of matches extracted by the `rex` command.
```

`plugins.ppl.subsearch.maxout` | `10000` | Sets the maximum of rows to return from subsearch.
`plugins.ppl.join.subsearch_maxout` | `50000` | Sets the maximum of rows from subsearch to join against.

> `plugins.ppl.pattern.method` | `"simple_pattern"` | Sets the default patterns method. Another supported method is "brain".

Author: It's quite complex to explain there method in a table. How about add a link to https://github.com/opensearch-project/sql/blob/main/docs/user/ppl/cmd/patterns.rst#syntax?

> `plugins.ppl.pattern.mode` | `"label"` | Sets the default patterns mode. Another supported mode is "aggregation".

Reviewer 2: Same comment - please remove quotation marks, list both valid values, and describe them so users can uderstand what they do.

Author: ditto

> `plugins.ppl.pattern.mode` | `"label"` | Sets the default patterns mode. Another supported mode is "aggregation".

Reviewer 2: ```suggestion
`plugins.ppl.pattern.mode` | `label` | Sets the mode for the `patterns` command. Valid values are `label` and `aggregation`. For more information, see [`patterns` syntax](https://github.com/opensearch-project/sql/blob/main/docs/user/ppl/cmd/patterns.rst#syntax).
```
